llvm/lib/Target/RISCV/RISCVDeadRegisterDefinitions.cpp

```diff
@@ -89,8 +96,8 @@ bool RISCVDeadRegisterDefinitions::runOnMachineFunction(MachineFunction &MF) {
           LLVM_DEBUG(dbgs() << "    Ignoring, register is not a GPR.\n");
           continue;
         }
+        LIS.removeInterval(Reg);
```

Comment:
nit: maybe could add `if(LIS.hasInterval(Reg))` or `assert(LIS.hasInterval(Reg))` to make sure the VReg has intervals.